### ISSUE TITLE
Cannot save image of an elemental map on Windows

### ISSUE THREAD

**Issue by jat255** [hyperspy/hyperspyUI#154]
### Steps to reproduce

Using the `hyperspy==1.3` and the latest master of `hyperspyui` (or the versions from the Windows bundle), there is no response when trying to call `File -> Save figure` on the results of an EDS elemental mapping (no console or terminal output). The EDS map is obtained by loading a signal with elemental information, and then clicking "Map" under the Element Picker widget.

This works on Linux, for what it's worth, but not on Windows when installed following the instructions in the README.

### Excpected behavior

A dialogue calling for the figure name should be brought up and then the figure should be saved.

**Comment by jat255** [comment]
Since this issue was never actively resolved (https://github.com/hyperspy/hyperspyUI/issues/118), I think it's the same problem.

**Comment by jat255** [comment]
Here is the (lack of) console output from opening HyperSpyUI, loading a signal, plotting the element map, and selecting "Save figure":

```
C:\Users\jat\git_repos
[hyperspyUI_test] λ python -m hyperspyui
INFO:hyperspy.io_plugins.bcf:The fast cython based bcf unpacking library were found
DEBUG:hyperspyui:Initializing plugins
DEBUG:hyperspyui:Found plugins: [<class 'hyperspyui.plugins.align.AlignPlugin'>, <class 'hyperspyui.plugins.alignzlp.Alignzlp'>, <class 'hyperspyui.plugins.axesconf.AxesConf'>, <class 'hyperspyui.plugins.axesorderwidget.AxesOrderPlugin'>, <class 'hyperspyui.plugins.basicsignal.BasicSignalPlugin'>, <class 'hyperspyui.plugins.basicspectrum.BasicSpectrumPlugin'>, <class 'hyperspyui.plugins.cmappicker.CMapPickerPlugin'>, <class 'hyperspyui.plugins.croptool.CropToolPlugin'>, <class 'hyperspyui.plugins.dmannotations.DmAnnotations'>, <class 'hyperspyui.plugins.eelsdb.EELSDBPlugin'>, <class 'hyperspyui.plugins.fft.FFT_Plugin'>, <class 'hyperspyui.plugins.fitting.FittingPlugin'>, <class 'hyperspyui.plugins.gaussianfilter.GaussianFilter'>, <class 'hyperspyui.plugins.imagerotation.ImageRotation_Plugin'>, <class 'hyperspyui.plugins.linemeasure.LineMeasure'>, <class 'hyperspyui.plugins.mva.MVA_Plugin'>, <class 'hyperspyui.plugins.metadataeditor.MetadataEditor'>, <class 'hyperspyui.plugins.mirrorplot.MirrorPlotPlugin'>, <class 'hyperspyui.plugins.moviesaver.MovieSaver'>, <class 'hyperspyui.plugins.plotutils.PlotUtils'>, <class 'hyperspyui.plugins.rebin.RebinPlugin'>, <class 'hyperspyui.plugins.recorderwidget.RecorderWidgetPlugin'>, <class 'hyperspyui.plugins.dataastext.SaveDataAsText'>, <class 'hyperspyui.plugins.segmentation.Segmentation'>, <class 'hyperspyui.plugins.stylesheet.StylePlugin'>, <class 'hyperspyui.plugins.tightlayout.Tightlayout'>, <class 'hyperspyui.plugins.gitgetter.GitSelector'>, <class 'hyperspyui.plugins.virtual_aperture.VirtualBfDf'>]
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.align.AlignPlugin'>
DEBUG:hyperspyui:Plugin loaded: Align
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.alignzlp.Alignzlp'>
DEBUG:hyperspyui:Plugin loaded: AlignZLP
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.axesconf.AxesConf'>
DEBUG:hyperspyui:Plugin loaded: Axes Configuration
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.axesorderwidget.AxesOrderPlugin'>
DEBUG:hyperspyui:Plugin loaded: Axes order widget
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.basicsignal.BasicSignalPlugin'>
DEBUG:hyperspyui:Plugin loaded: Basic signal tools
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.basicspectrum.BasicSpectrumPlugin'>
DEBUG:hyperspyui:Plugin loaded: Basic spectrum tools
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.cmappicker.CMapPickerPlugin'>
DEBUG:hyperspyui:Plugin loaded: Color map picker
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.croptool.CropToolPlugin'>
DEBUG:hyperspyui:Plugin loaded: Crop tool
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.dmannotations.DmAnnotations'>
DEBUG:hyperspyui:Plugin loaded: DM Annotations
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.eelsdb.EELSDBPlugin'>
DEBUG:hyperspyui:Plugin loaded: EELSDB Plugin
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.fft.FFT_Plugin'>
DEBUG:hyperspyui:Plugin loaded: FFT
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.fitting.FittingPlugin'>
DEBUG:hyperspyui:Plugin loaded: Fitting plugin
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.gaussianfilter.GaussianFilter'>
DEBUG:hyperspyui:Plugin loaded: Gaussian Filter
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.imagerotation.ImageRotation_Plugin'>
DEBUG:hyperspyui:Plugin loaded: Image Rotation
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.linemeasure.LineMeasure'>
DEBUG:hyperspyui:Plugin loaded: LineMeasure
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.mva.MVA_Plugin'>
DEBUG:hyperspyui:Plugin loaded: MVA
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.metadataeditor.MetadataEditor'>
DEBUG:hyperspyui:Plugin loaded: Metadata Editor
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.mirrorplot.MirrorPlotPlugin'>
DEBUG:hyperspyui:Plugin loaded: Mirror plot
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.moviesaver.MovieSaver'>
DEBUG:hyperspyui:Plugin loaded: Movie Saver
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.plotutils.PlotUtils'>
DEBUG:hyperspyui:Plugin loaded: Plot utils
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.rebin.RebinPlugin'>
DEBUG:hyperspyui:Plugin loaded: Rebin
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.recorderwidget.RecorderWidgetPlugin'>
DEBUG:hyperspyui:Plugin loaded: Recorder Widget
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.dataastext.SaveDataAsText'>
DEBUG:hyperspyui:Plugin loaded: Save data as text
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.segmentation.Segmentation'>
DEBUG:hyperspyui:Plugin loaded: Segmentation
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.stylesheet.StylePlugin'>
DEBUG:hyperspyui:Plugin loaded: Style
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.tightlayout.Tightlayout'>
DEBUG:hyperspyui:Plugin loaded: TightLayout
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.gitgetter.GitSelector'>
DEBUG:hyperspyui:Plugin loaded: Version selector
DEBUG:hyperspyui:Initializing plugin: <class 'hyperspyui.plugins.virtual_aperture.VirtualBfDf'>
DEBUG:hyperspyui:Plugin loaded: Virtual BF/DF
DEBUG:hyperspyui:Creating default actions
DEBUG:hyperspyui:Creating plugin actions
DEBUG:hyperspyui:Creating console
DEBUG:hyperspyui:Creating menus
DEBUG:hyperspyui:Creating toolbars
DEBUG:hyperspyui:Creating widgets
INFO:hyperspyui:Main window loaded!
```

**Comment by jat255** [comment]
Figured out the problem:

Because of how we auto-name the elemental maps with `get_eds_line_intensity()`, there is a newline character, and a colon, which are not handled elegantly by Windows at https://github.com/hyperspy/hyperspyUI/blob/master/hyperspyui/mainwindowutillayer.py#L377.

I think the best bet would be to somehow sanitize the "suggested" path of illegal characters before calling that method.

**Comment by thomasaarholt** [comment]
Nice catch!

**Comment by jat255** [comment]
This is only partially resolved by #150. When using the qt5 support, the save figure window will open, but the user will be unable to click "Save" without modifying the filename because it contains `:` and `\n`
